Flexa-Agents/large-withdrawal/src/agent.spec.ts

```diff
@@ -0,0 +1,146 @@
+
+const transferByPartitionFinding = (fromPartition: string, fromAddress: string, toAddress: string, value: string): Finding => Finding.fromObject({
+  name: "Large transfer from a Flexa Collateral Manager partition detected",
+  description: "TransferByPartition event emitted with a large value",
+  alertId: "FLEXA-2",
+  severity: FindingSeverity.Info,
+  type: FindingType.Info,
+  protocol: "Flexa",
+  metadata: { fromPartition, fromAddress, toAddress, value },
+});
+
+describe("Forta Collateral Manager large transfer by partition agent test suite", () => {
+    const mockIsPartition = jest.fn();
+    const FLEXA_CONTRACT: string = createAddress("0xfffd");
+    const AMP_CONTRACT: string = createAddress("0xffff");
+    const mockFetcher = {
+        isPartition: mockIsPartition,
+        flexa: FLEXA_CONTRACT,
+        amp: AMP_CONTRACT
+    }
+    const THRESHOLD: BigNumber = BigNumber.from("1000");
+    const handler: HandleTransaction = provideHandleTransaction(mockFetcher as any, THRESHOLD);
+
+    beforeEach(() => {
+        mockFetcher.isPartition.mockClear();
+    });
+
+    it("should report no findings for txs without events", async () => {
+        const txEvent: TransactionEvent = new TestTransactionEvent();
+        const findings: Finding[] = await handler(txEvent);
+        expect(findings).toStrictEqual([]);           
+    });
+
+    it("should ignore regular, below the threshold, transfers", async () => {
+        when(mockIsPartition)
+            .calledWith(38, PARTITIONS[1]).mockReturnValue(true);
+            
+        const event = AMP_IFACE.getEvent("TransferByPartition");
+        const log = AMP_IFACE.encodeEventLog(
+            event, [PARTITIONS[1], createAddress('0x4567'),  mockFetcher.flexa, createAddress("0x7676"), "324", keccak256("k6543"), keccak256("d4949d")]
+        );
+        
+        const txEvent: TransactionEvent =  new TestTransactionEvent()
+            .setBlock(38)
+            .addAnonymousEventLog(mockFetcher.amp, log.data, ...log.topics);        
+
+        const findings: Finding[] = await handler(txEvent);
+        expect(findings).toStrictEqual([]);
+
+    })
+
+    it("should ignore non-withdrawal transfers", async () => {
+        when(mockIsPartition)
+            .calledWith(22, PARTITIONS[4]).mockReturnValue(true);
+            
+        const event = AMP_IFACE.getEvent("TransferByPartition");
+        const log = AMP_IFACE.encodeEventLog(
+            event, [PARTITIONS[4], createAddress('0x3333'),  createAddress("0x4555"), createAddress("0x4444"), "155324", keccak256("accbbb"), keccak256("ddeedd")]
+        );
+        
+        const txEvent: TransactionEvent =  new TestTransactionEvent()
+            .setBlock(22)
+            .addAnonymousEventLog(mockFetcher.amp, log.data, ...log.topics)        
+
+        const findings: Finding[] = await handler(txEvent);
+        expect(findings).toStrictEqual([]);
+    })
+
+    it("should ignore transfers not from a partition", async () => {
```

Comment:
```suggestion
    it("should ignore transfers from a non-permitted partition", async () => {
```
Slight name change suggestion to be more specific.

Reply:
Thanks - Should the function `isPartition` be changed to `isPermittedPartition` (and var `isFlexaPartition` to `isPermittedFlexaPartition`) as well ?

Reply:
I think those can stay as they are, I only suggested the name change for the unit test because it took me a second to figure out what it was testing.